### PR TITLE
Code quality

### PR DESCRIPTION
**Pull request by DigitalTimK**
Improve code quality
- apply php7.4 style (remove redundant annotations)
- bring code to phpstan level 8
- corrections of test to make them work
- updated file-comment (license remark) to year 2024

Enjoy reading 😜

### CODE REVIEW

**Comment by GhaziTriki** [comment]
Excellent contribution @DigitalTimK. I have some good reading for tonight.

**Comment by andrekutianski** [comment]
Considering the large code refactor, is it wise to postpone the merge to pull new features?

**Comment by DigitalTimK** [comment]
Hello Andre,

> Considering the large code refactor, is it wise to postpone the merge to pull new features?

With that argumentation, any new PR will gets a higher priority than this PR. At the end the code has changed sooo much, that this PR will be obsolete. No, I guess it is wise to get this PR prioritized and progressed as soon as possible. 

I am also waiting for the approval of this PR to continue with new improvements...

@GhaziTriki: What is you plan to progress? Can we support?

File: src/Util/FileDumper.php
Comment (GhaziTriki) [inline src/Util/FileDumper.php:31]: The API library has currently no access to the file system. Dealing with file system exposes the end platform to several security risks. Where do we need this in the library? If it is intended for debugging I suggest using `tracy` library in require-dev.

File: src/Core/Format.php
Comment (GhaziTriki) [inline src/Core/Format.php:45]: Trade-off between more memory (old version) and more cpu (new version). Need a good argument to decided which solution to choose for the community. We assume the XML content is static and never changes in a single API call.

File: src/Core/Record.php
Comment (GhaziTriki) [inline src/Core/Record.php:149]: Trade-off between more memory (old version) and more cpu (new version). Need a good argument to decided which solution to choose for the community. We assume the XML content is static and never changes in a single API call.

File: src/Parameters/JoinMeetingParameters.php
Comment (GhaziTriki) [inline src/Parameters/JoinMeetingParameters.php:43]: It might break existing integrations storing this value in the database without float transformation but not sure.

File: src/Responses/GetMeetingsResponse.php
Comment (GhaziTriki) [inline src/Responses/GetMeetingsResponse.php:35]: Trade-off between more memory (old version) and more cpu (new version). Need a good argument to decided which solution to choose for the community. We assume the XML content is static and never changes in a single API call.

File: src/Responses/GetRecordingTextTracksResponse.php
Comment (GhaziTriki) [inline src/Responses/GetRecordingTextTracksResponse.php:35]: Trade-off between more memory (old version) and more cpu (new version). Need a good argument to decided which solution to choose for the community. We assume the XML content is static and never changes in a single API call.

File: src/Responses/GetRecordingsResponse.php
Comment (GhaziTriki) [inline src/Responses/GetRecordingsResponse.php:36]: Trade-off between more memory (old version) and more cpu (new version). Need a good argument to decided which solution to choose for the community. We assume the XML content is static and never changes in a single API call.

File: src/Responses/HooksListResponse.php
Comment (GhaziTriki) [inline src/Responses/HooksListResponse.php:36]: Trade-off between more memory (old version) and more cpu (new version). Need a good argument to decided which solution to choose for the community. We assume the XML content is static and never changes in a single API call.

**Review by GhaziTriki — CHANGES_REQUESTED** [review]

File: .gitignore
Comment (DigitalTimK) [reply .gitignore:3]: mine too

File: composer.json
Comment (DigitalTimK) [reply composer.json:30]: yes, to my knowledge it is needed for `json_encode()` and `json_decode()`. As it is not 100% ensured that is installed on the system I added this line. With php8 it can be removed again ;-)

File: tests/Parameters/PublishRecordingsParametersTest.php
Comment (DigitalTimK) [reply tests/Parameters/PublishRecordingsParametersTest.php:36]: I thought that it would give enough meaning ;-) What do you have in mind?

I adapted to this:
```php
    public function testPublishRecordingsParameters()
    {
        $recordingId1 = $this->faker->uuid;
        $publish1     = $this->faker->boolean(50);
        $recordingId2 = $this->faker->uuid;
        $publish2     = !$publish1;

        $publishRecording = new PublishRecordingsParameters($recordingId1, $publish1);

        $this->assertEquals($recordingId1, $publishRecording->getRecordingId());
        $this->assertEquals($publish1, $publishRecording->isPublish());

        // Test setters that are ignored by the constructor
        $publishRecording->setRecordingId($recordingId2);
        $publishRecording->setPublish($publish2);
        $this->assertEquals($recordingId2, $publishRecording->getRecordingId());
        $this->assertEquals($publish2, $publishRecording->isPublish());
    }
```

Do you agree?

File: src/BigBlueButton.php
Comment (DigitalTimK) [reply src/BigBlueButton.php:411]: I just improved that function by re-grouping its elements. Not sure if this is covered by composer. I would avoid changing it now and keep it to be 100% on the save side ... otherwise to much refactoring at once ...

File: src/BigBlueButton.php
Comment (DigitalTimK) [reply src/BigBlueButton.php:470]: see `curl_exec()`-[docs](https://www.php.net/manual/en/function.curl-exec.php)

> **Return** bool|string true on success or false on failure. However, if the CURLOPT_RETURNTRANSFER option is set, it will return the result on success, false on failure

File: src/BigBlueButton.php
Comment (DigitalTimK) [reply src/BigBlueButton.php:474]: You mean the status code check? Not sure. I was just moving logically elements. The status check was already introduced on July, 30th 2020.

But it makes sense, since 400-codes are the good once ;-)

File: src/Util/FileDumper.php
Comment (DigitalTimK) [reply src/Util/FileDumper.php:31]: yes, the only reason was debugging. 
I can remove the class (and the`var`in `.gitignore` if you don't like it. And I do not know `tracy`... that maybe the reason i needed that class ;-)

File: src/Core/Record.php
Comment (DigitalTimK) [reply src/Core/Record.php:149]: Technically you are right (memory vs. cpu), but I guess those API-small data do no harm to any of them in these days. This proposal is just a cleaner and leaner way ... and it is more robust against programming errors (less code, less variables, low risk of inconsistent data).

File: src/Responses/GetMeetingsResponse.php
Comment (DigitalTimK) [reply src/Responses/GetMeetingsResponse.php:35]: Technically you are right (memory vs. cpu), but I guess those API-small data do no harm to any of them in these days. This proposal is just a cleaner and leaner way ... and it is more robust against programming errors (less code, less variables, low risk of inconsistent data).

File: src/Responses/HooksListResponse.php
Comment (DigitalTimK) [reply src/Responses/HooksListResponse.php:36]: Technically you are right (memory vs. cpu), but I guess those API-small data do no harm to any of them in these days. This proposal is just a cleaner and leaner way ... and it is more robust against programming errors (less code, less variables, low risk of inconsistent data).

File: src/Responses/GetRecordingTextTracksResponse.php
Comment (DigitalTimK) [reply src/Responses/GetRecordingTextTracksResponse.php:35]: Technically you are right (memory vs. cpu), but I guess those API-small data do no harm to any of them in these days. This proposal is just a cleaner and leaner way ... and it is more robust against programming errors (less code, less variables, low risk of inconsistent data).

File: src/Parameters/JoinMeetingParameters.php
Comment (DigitalTimK) [reply src/Parameters/JoinMeetingParameters.php:43]: The usage of `creationTime` was very in consistent before, sometimes `string`, sometimes `int` and in the PhpDoc (src/Responses/CreateMeetingResponse.php) it was mentioned float. The XML comes as `string` and must anyway be transformed into `int` or `float`. The usage inside of bigbluebutton-api is now consistent as `float`. I decided for `float` due to the [x-day issue of the UNIX-time](https://www.leebutterman.com/2021/02/01/store-your-unix-epoch-times-as-float64.html).

Since it was inconsistent before, we will break anyhow something... but now it is "technically" I guess the correct approach.

File: composer.json
Comment (GhaziTriki) [reply composer.json:30]: https://www.php.net/manual/en/json.installation.php

File: src/Util/FileDumper.php
Comment (GhaziTriki) [reply src/Util/FileDumper.php:31]: If we really need debugging for dev, I suggest adding https://github.com/nette/tracy to dev dependencies. `FileDumper` **must** be removed (sorry to say like it that) as it exposes too many risks.

File: tests/Parameters/DeleteRecordingsParametersTest.php
Comment (DigitalTimK) [reply tests/Parameters/DeleteRecordingsParametersTest.php:40]: some here... not sure what kind of name you like to see...

Find below my new proposal. Ok?

```php
    public function testDeleteRecordingParameter()
    {
        $recordingId1 = $this->faker->uuid;
        $recordingId2 = $this->faker->uuid;

        // Test by constructor
        $deleteRecording = new DeleteRecordingsParameters($recordingId1);
        $this->assertEquals($recordingId1, $deleteRecording->getRecordingId());

        // Test setters that are ignored by the constructor
        $deleteRecording->setRecordingId($recordingId2);
        $this->assertEquals($recordingId2, $deleteRecording->getRecordingId());
    }
```

File: .gitignore
Comment (DigitalTimK) [reply .gitignore]: `var`-folder removed from `.gitignore` as not needed anymore

File: tests/Parameters/PublishRecordingsParametersTest.php
Comment (DigitalTimK) [reply tests/Parameters/PublishRecordingsParametersTest.php]: This version implemented in new commit.

File: tests/Parameters/DeleteRecordingsParametersTest.php
Comment (DigitalTimK) [reply tests/Parameters/DeleteRecordingsParametersTest.php]: This version implemented in new commit.

**Pull request merged**
